ebpf/monitor/accesslog.proto

```diff
@@ -0,0 +1,340 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+syntax = "proto3";
+
+package skywalking.v3;
+
+option java_multiple_files = true;
+option java_package = "org.apache.skywalking.apm.network.ebpf.monitor.accesslog.v3";
+option go_package = "skywalking.apache.org/repo/goapi/collect/ebpf/monitor/accesslog/v3";
+
+import "common/Common.proto";
+import "common/Command.proto";
+
+service EBPFAccessLogService {
+  rpc collect(stream EBPFAccessLogMessage) returns (EBPFAccessLogDownstream) {
+  }
+}
+
+message EBPFAccessLogMessage {
+  // current node information, only not null when first message or have update
+  EBPFAccessLogNodeInfo node = 1;
+  // local process and remote process connection information
+  AccessLogConnection connection = 2;
+  // kernel level metrics
+  repeated AccessLogKernelLog kernelLogs = 3;
+  // application protocol log
+  // if the protocol is detected, the kernel logs is works the related logs
+  // otherwise, the kernel log not related and send with frequency
+  AccessLogProtocolLogs protocolLog = 4;
+}
+
+message EBPFAccessLogNodeInfo {
+  // Node name
+  string name = 1;
+  // All net interfaces list
+  repeated EBPFAccessLogNodeNetInterface netInterfaces = 2;
+  // System boot time
+  Instant bootTime = 3;
+}
+
+message EBPFAccessLogNodeNetInterface {
+  int32 index = 1;
+  int32 mtu = 2;
+  string name = 3;
+}
+
+// Connection information
+message AccessLogConnection {
+  // local address
+  ConnectionAddress local = 1;
+  // remote/peer address
+  ConnectionAddress remote = 2;
+  // local address role
+  AccessLogConnectionRole role = 3;
+  // is the connection using TLS or not
+  AccessLogConnectionTLSMode tlsMode = 4;
+}
+
+message ConnectionAddress {
+  oneof address {
+    // if the address is monitored under the local machine, then return the kubernetes
+    KubernetesProcessAddress kubernetes = 1;
+    // if the address cannot be aware, then return the ip address
+    IPAddress ip = 2;
+  }
+}
+
+message KubernetesProcessAddress {
+  string serviceName = 1;
+  string podName = 2;
+  string containerName = 3;
+  string processName = 4;
+  int32 port = 5;
+}
+
+message IPAddress {
+  string host = 1;
+  int32 port = 2;
+}
+
+enum AccessLogConnectionRole {
+  Client = 0;
+  Server = 1;
+}
+
+enum AccessLogConnectionTLSMode {
+  Normal = 0;
+  TLS = 1;
+}
```

Comment:
Do we have status of tls, mtls, etc.?

Reply:
Nope, for now, the eBPF only can detect TLS.